nextroute/schema/output.go

```diff
@@ -6,60 +6,96 @@ import (
+	Route []PlannedStopOutput `json:"route"`
+	// RouteTravelDuration is the total travel duration of the vehicle.
+	RouteTravelDuration int `json:"route_travel_duration"`
+	// RouteTravelDistance is the total travel distance of the vehicle.
+	RouteTravelDistance int `json:"route_travel_distance,omitempty"`
+	// RouteStopsDuration is the total stops duration of the vehicle.
+	RouteStopsDuration int `json:"route_stops_duration,omitempty"`
+	// RouteWaitingDuration is the total waiting duration of the vehicle.
+	RouteWaitingDuration int `json:"route_waiting_duration,omitempty"`
+	// RouteDuration is the total duration of the vehicle.
+	RouteDuration int `json:"route_duration"`
+	// CustomData is the custom data of the vehicle.
+	CustomData any `json:"custom_data,omitempty"`
+	// AlternateStops is the list of alternate stops selected.
+	AlternateStops *[]string `json:"alternate_stops,omitempty"`
 }
 
 // PlannedStopOutput adds information to the input stop.
 type PlannedStopOutput struct {
-	Stop                     StopOutput `json:"stop"`
-	TravelDuration           int        `json:"travel_duration"`
-	CumulativeTravelDuration int        `json:"cumulative_travel_duration"`
-	TravelDistance           int        `json:"travel_distance,omitempty"`
-	CumulativeTravelDistance int        `json:"cumulative_travel_distance,omitempty"`
-	TargetArrivalTime        *time.Time `json:"target_arrival_time,omitempty"`
-	ArrivalTime              *time.Time `json:"arrival_time,omitempty"`
-	WaitingDuration          int        `json:"waiting_duration,omitempty"`
-	StartTime                *time.Time `json:"start_time,omitempty"`
-	Duration                 int        `json:"duration,omitempty"`
-	EndTime                  *time.Time `json:"end_time,omitempty"`
-	EarlyArrivalDuration     int        `json:"early_arrival_duration,omitempty"`
-	LateArrivalDuration      int        `json:"late_arrival_duration,omitempty"`
-	CustomData               any        `json:"custom_data,omitempty"`
+	// Stop is the input stop.
+	Stop StopOutput `json:"stop"`
+	// TravelDuration is the travel duration of the stop.
+	TravelDuration int `json:"travel_duration"`
+	// CumulativeTravelDuration is the total travel duration to get to this location.
+	CumulativeTravelDuration int `json:"cumulative_travel_duration"`
+	// TravelDistance is the travel distance of the stop.
+	TravelDistance int `json:"travel_distance,omitempty"`
+	// CumulativeTravelDistance is the total travel distance to get to this location.
+	CumulativeTravelDistance int `json:"cumulative_travel_distance,omitempty"`
+	// TargetArrivalTime is the target arrival time of the stop.
+	TargetArrivalTime *time.Time `json:"target_arrival_time,omitempty"`
+	// ArrivalTime is the arrival time of the stop.
+	ArrivalTime *time.Time `json:"arrival_time,omitempty"`
+	// WaitingDuration is the waiting duration of the stop in seconds.
+	WaitingDuration int `json:"waiting_duration,omitempty"`
+	// StartTime is the start time of the stop.
```

Comment:
```suggestion
	// StartTime is the time the vehicle starts servicing this stop.
```
Start time sounds a bit imprecise to me (it could be that we count driving towards the stop already as part of starting "work" on the stop or so). I am not sure whether we like servicing though. Maybe there is a better option? :shrug: